File: briefcase/src/types/asset.ts
export type Importance =
  | "critical"
  | "executive"
  | "technical"
  | "research"
  | "approved"
  | "archived";

export type Audience =
  | "executive"
  | "engineering"
  | "client"
  | "internal"
  | "public";

export type Confidentiality = "confidential" | "internal" | "unrestricted";

export type VettedStatus =
  | "unvetted"
  | "ai-vetted"
  | "peer-reviewed"
  | "human-approved"
  | "final-locked";

export type RelationshipType =
  | "derived-from"
  | "supersedes"
  | "presented-in"
  | "reviewed-by"
  | "rendered-from"
  | "exported-as"
  | "bundles"
  | "references";

export interface AssetRecord {
  id: string;
  contentHash: string;
  title: string;
  fileType: string;
  originalName: string;
  project: string | null;
  subproject: string | null;
  sourceAgent: string | null;
  pipelineId: string | null;
  importance: Importance;
  audience: Audience;
  confidentiality: Confidentiality;
  vettedStatus: VettedStatus;
  tags: string[];
  classificationConfidence: number;
  classificationExplanation: string;
  createdAt: string;
  ingestedAt: string;
  version: number;
  supersededBy: string | null;
}

export interface AssetRelationship {
  sourceId: string;
  targetId: string;
  relationshipType: RelationshipType;
  createdAt: string;
}

// UI grouping helper
export interface DrawerGroup {
  key: string;
  label: string;
  importance?: Importance;
  assetIds: string[];
  isOpen: boolean;
}

File: briefcase/src-tauri/src/models/asset.rs
use serde::{Deserialize, Serialize};
use super::enums::{Audience, Confidentiality, Importance, VettedStatus};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AssetRecord {
    pub id: String,
    pub content_hash: String,
    pub title: String,
    pub file_type: String,
    pub original_name: String,
    pub project: Option<String>,
    pub subproject: Option<String>,
    pub source_agent: Option<String>,
    pub pipeline_id: Option<String>,
    pub importance: Importance,
    pub audience: Audience,
    pub confidentiality: Confidentiality,
    pub vetted_status: VettedStatus,
    pub tags: Vec<String>,
    pub classification_confidence: f64,
    pub classification_explanation: String,
    pub created_at: String,
    pub ingested_at: String,
    pub version: i64,
    pub superseded_by: Option<String>,
}

impl AssetRecord {
    pub fn tags_json(&self) -> String {
        serde_json::to_string(&self.tags).unwrap_or_else(|_| "[]".to_string())
    }
}
